src/test/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceServiceSpec.scala

```diff
@@ -766,6 +766,30 @@ class ResourceServiceSpec extends FlatSpec with Matchers with ScalaFutures with
     assert(policyDAO.listAccessPolicies(parentResource, samRequestContext).unsafeRunSync().nonEmpty)
   }
 
+  "validatePolicy" should "succeed with a correct policy" in {
+    runAndWait(service.validatePolicy(defaultResourceType, service.ValidatableAccessPolicy(AccessPolicyName("a"), Map(dummyUserInfo.userEmail -> Option(dummyUserInfo.userId.asInstanceOf[WorkbenchSubject])), Set(ResourceRoleName("owner")), Set(ResourceAction("alter_policies")), Set()))) shouldBe empty
+  }
+
+  "validatePolicy" should "succeed with an incorrect policy" in {
+    runAndWait(service.validatePolicy(defaultResourceType, service.ValidatableAccessPolicy(AccessPolicyName("a"), Map(dummyUserInfo.userEmail -> Option(dummyUserInfo.userId.asInstanceOf[WorkbenchSubject])), Set(ResourceRoleName("bad_name")), Set(ResourceAction("bad_action")), Set()))) shouldBe defined
```

Comment:
Would like to see this also assert that the resulting `ErrorReport` is for the right reason.

Same for the other tests that expect the validation to fail.